zerver/views/streams.py

```diff
@@ -256,17 +256,18 @@ def remove_subscriptions_backend(
     removing_someone_else = principals and \
         set(principals) != {user_profile.email}
 
-    if removing_someone_else and not user_profile.is_realm_admin:
-        # You can only unsubscribe other people from a stream if you are a realm
-        # admin (whether the stream is public or private).
-        return json_error(_("This action requires administrative rights"))
-
     streams_as_dict = []
     for stream_name in streams_raw:
         streams_as_dict.append({"name": stream_name.strip()})
 
     streams, __ = list_to_streams(streams_as_dict, user_profile)
 
+    if removing_someone_else:
+        for stream in streams:
+            # Only stream admin and realm admins can unsubscribe
+            # other users from a stream.
+            access_stream_for_admin_actions(user_profile, stream)
```

Comment:
This change is inefficient, adding database queries in a loop to this code path.  I think we want to make `list_to_streams` support doing this check if a `require_stream_admin=True` option passed, since it's doing proper bulk queries.

Reply:
`list_to_streams` calls `bulk_get_streams` to get all the streams, so I am unable to understand clearly how to make `access_stream..` to return a stream object as you suggested in one of the above comments. Any suggestions would be helpful.